openhands/memory/condenser/impl/llm_summarizing_condenser.py

```diff
@@ -32,26 +36,22 @@ def __init__(self, llm: LLM, max_size: int = 100, keep_first: int = 1):
 
         super().__init__()
 
-    def condense(self, events: list[Event]) -> list[Event]:
-        """Apply the amortized forgetting strategy with LLM summarization to the given list of events."""
-        if len(events) <= self.max_size:
-            return events
-
-        head = events[: self.keep_first]
-
+    def get_condensation(self, view: View) -> Condensation:
+        head = view[: self.keep_first]
         target_size = self.max_size // 2
-        events_from_tail = target_size - len(head)
-        tail = events[-events_from_tail:]
+        # Number of events to keep from the tail -- target size, minus however many
+        # prefix events from the head, minus one for the summarization event
+        events_from_tail = target_size - len(head) - 1
 
         summary_event = (
-            events[self.keep_first]
-            if isinstance(events[self.keep_first], AgentCondensationObservation)
+            view[self.keep_first]
+            if isinstance(view[self.keep_first], AgentCondensationObservation)
```

Comment:
```suggestion
            if isinstance(view[self.keep_first], CondensationAction)
```
It's an Action, in this PR, right?

Reply:
Different events for different uses. The action tells us to condense something in the history, the observation is the summary we send to the LLM.

Reply:
> The action has the summary, though, of course, it needs to have it.

The action has the summary, yes, but we don't have access to the actions while computing the next condensation, just the view that was produced. We could also pass the list of condensation actions in as a field of the `View` data class but I don't know that it helps clean things up much.

> Not for this PR IMHO, one alternative could be to remove the use of the obs in condensers and keep it only for the emergency truncation - as you were suggesting before, to separate the two.

The observation is what actually gets passed to the LLM, though. I'm happy to rename the observation to make things more clear (we inherited the observation name from the OG condenser implementation), but they are distinct events and shouldn't be made the same.

Reply:
I see, thank you. You may be right! 🤔If it keeps things manageable in condensers, maybe it's not an issue... We could also do something else with the truncation obs.